gsl/span

```diff
@@ -163,6 +163,8 @@ namespace details
         {
         }
 
+        constexpr span_iterator<Span, IsConst>& operator=(const span_iterator<Span, IsConst>&) = default;
```

Comment:
@garyfurnish is correct, it is intentional to allow conversions between the two types of iterator.